File: projects/order-simbaconnector-bc/src/main/java/com/robinfood/dtos/request/orderbill/ReplacementPortionDTO.java
package com.robinfood.dtos.request.orderbill;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@Builder
@Data
@NoArgsConstructor
public class ReplacementPortionDTO {

    private Long id;

    private String name;

    private PortionProductDTO product;

    private String sku;

    private Long unitId;

    private Double unitNumber;

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/ReplacementPortionDTO.kt
package com.robinfood.core.dtos.transactionrequest

import javax.validation.Valid
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull
import javax.validation.constraints.Positive

class ReplacementPortionDTO (

    @NotNull
    @Positive
    val id: Long,

    @NotBlank
    val name: String,

    @NotNull
    @Valid
    val product: PortionProductDTO,

    @NotBlank
    val sku: String,

    @NotNull
    @Positive
    val unitId: Long,

    @NotNull
    @Positive
    val unitNumber: Double
)
